get-poetry.py

```diff
@@ -194,7 +195,7 @@ def expanduser(path):
 POETRY_LIB_BACKUP = os.path.join(POETRY_HOME, "lib-backup")
 
 
-BIN = """#!/usr/bin/env python
+BIN = """#!/usr/bin/env {python_exe}
```

Comment:
I would simple replace BIN with `f"#!{sys.executable}"`. What is the purpose of using env ? Why use project python to run poetry ?

Reply:
I don't know if replacing `BIN` with `f"#!{sys.executable}"` could break the behavior.

Currently, `poetry install` correctly install the Python packages to pre-selected virtual env. If change to `f"#!{sys.executable}"`, I wonder if that behaviour can still be kept.

Reply:
```
> /home/kamal/.local/pipx/venvs/poetry/lib/python3.7/site-packages/poetry/console/commands/install.py(52)handle()
-> extras = []
(Pdb) l
 47  	        installer = Installer(
 48  	            self.io, self.env, self.poetry.package, self.poetry.locker, self.poetry.pool
 49  	        )
 50  	        import pdb;pdb.set_trace()
 51  	
 52  ->	        extras = []
 53  	        for extra in self.option("extras"):
 54  	            if " " in extra:
 55  	                extras += [e.strip() for e in extra.split(" ")]
 56  	            else:
 57  	                extras.append(extra)
(Pdb) self.env
VirtualEnv("/home/kamal/python/demo/.venv")
(Pdb) import sys
(Pdb) sys.executable
'/home/kamal/.local/pipx/venvs/poetry/bin/python'
```
As we can see above, the installer already running with correct venv context, independent of `sys.executable`.

Reply:
Can confirm that this small delta fixes the script:

```diff
--- get-poetry.py       2020-03-25 17:06:28.451064345 -0400
+++ get-poetry.py.new   2020-03-25 17:04:19.746420009 -0400
@@ -197,7 +197,7 @@
 POETRY_LIB_BACKUP = os.path.join(POETRY_HOME, "lib-backup")
 
 
-BIN = """#!/usr/bin/env python
+BIN = f"""#!{sys.executable}
 # -*- coding: utf-8 -*-
 import glob
 import sys
```

along with invoking the script with `python3`, on Debian Buster (10.x).

Reply:
If so, I opt for `f"#!{sys.executable}"`. The only issue is that `f-string` can only qork with Python 3.6+.

I wonder if @sdispater want to maintain compatibility with Python 3.5. This is the default Python version of Debian Stretch (9).

Reply:
Right, in my parallel PR I moved back to a `.format()` approach, see https://github.com/python-poetry/poetry/pull/2222 .

Happy to close mine if this one has traction, but mine's passing all the tests right now.